logical/framework/path_map.go

```diff
@@ -73,17 +73,22 @@ func (p *PathMap) pathStruct(s logical.Storage, k string) (*PathStruct, error) {
 		Schema: p.Schema,
 	}
 
-	// Check for unsalted version and upgrade if so
-	if k != origKey {
-		// Generate the unsalted name
-		unsaltedName := fmt.Sprintf("map/%s/%s", p.Name, origKey)
-		// Set the path struct to use the unsalted name
-		ps.Name = unsaltedName
+	if !strings.HasPrefix(origKey, "s") && k != origKey {
```

Comment:
Should this be an OR instead of an AND?

Reply:
No -- that would actually lead to the same bad behavior, because in the read-with-list-value case `k` will not be equal to the original key since the original input is already hashed.